datasafe-simple-adapter/datasafe-simple-adapter-impl/src/main/java/de/adorsys/datasafe/simple/adapter/impl/SimpleDatasafeServiceImpl.java

```diff
@@ -69,13 +69,8 @@ public SimpleDatasafeServiceImpl(DFSCredentials dfsCredentials, MutableEncryptio
             log.info(lsf.toString());
             this.systemRoot = FileSystems.getDefault().getPath(filesystemDFSCredentials.getRoot()).toAbsolutePath().toUri();
             storageService = new FileSystemStorageService(FileSystems.getDefault().getPath(filesystemDFSCredentials.getRoot()));
-            customlyBuiltDatasafeServices = DaggerLegacyDatasafeService.builder()
+            customlyBuiltDatasafeServices = DaggerSwitchableDatasafeServices.builder()
```

Comment:
We should use MutableEncryptionConfig:
```
customlyBuiltDatasafeServices = DaggerSwitchableDatasafeServices.builder()
                    .config(new DefaultDFSConfig(systemRoot, universalReadStorePassword))
                    .storage(getStorageService())
                    .encryption(config.toEncryptionConfig())
                    .build();
```

Reply:
is removed with your next improvment

Reply:
Here MutableEncryption is simply encryption config POJO/DTO that comes from Spring application.properties
This:
```
encryption(config.toEncryptionConfig())
```
is technically equal to 
SwitchableDatasafeService.setEncryption(config);
with respect of the fact it is immutable

We provide external entity to Dagger